server/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -355,8 +355,7 @@ private static String changeFirstCharacterCase(String str, boolean capitalize) {
 
     public static boolean validFileName(String fileName) {
         for (int i = 0; i < fileName.length(); i++) {
-            char c = fileName.charAt(i);
-            if (INVALID_FILENAME_CHARS.contains(c)) {
```

Comment:
We should remove the Set? That can be a followup, but keeping it around is confusing.

Reply:
Did it here in https://github.com/elastic/elasticsearch/pull/85660/commits/421b5c60fbc6bd4ede3bbf4ee8bf08a46313a2cf the current set is only used to render an exception message in production and it rendered in a broken way (just inlined the char set in the one test using it, which seemed good enough to me).